### ISSUE TITLE
Update .editorconfig and rulesets based on feedback

### ISSUE THREAD

**Issue by originalfoo** [CitiesSkylinesMods/DuplicateAssemblyScanner#19]
Dymanoid mentioned following in PR #16:

* The .NET naming convention suggests using PascalCase for constants.
* The .NET naming convention suggests either PascalCase or camelCase for static fields. The underscore should be used as a prefix for _instanceFields.

These things should be configured in the .editorconfig and/or ruleset.

**Comment by dymanoid** [comment]
You can just use `StyleCop.Analyzers`. These rules are activated there by default, including the BSD-style of the braces and many more.

You will also have the Quick Fixes for those rules, so you can auto**magic**ally fix all code style violations in the whole solution by one mouse click.

**Comment by originalfoo** [comment]
My plan is to have a separate git repo for the ruleset and the .editorconfig and then include that in to all the others, that way any changes can be rolled out to all repos.

Currently no idea how to do that or even if it will work heh.

**Comment by dymanoid** [comment]
Currently, there is no built-in way to share `.editorconfig` and stuff across repositories. However, there are open issues in the C# Roslyn compiler repo here, on GitHub. It is currently in state "needs design" (see https://github.com/dotnet/roslyn/issues/19028).

There are [some attempts](https://stackoverflow.com/questions/52186841/editorconfig-in-shared-git-repository) to [create workarounds](https://stackoverflow.com/questions/58202037/is-it-possible-to-share-a-single-file-in-the-root-folder-of-multiple-git-reposit), but I doubt the effort is worth it, and these workarounds heavily depend on the client usage.

**Comment by leonardochaia** [comment]
> My plan is to have a separate git repo for the ruleset and the .editorconfig and then include that in to all the others, that way any changes can be rolled out to all repos.

Hi @aubergine10 , did you figure this out? I'm trying to figure out the best way to distribute .editorconfig files organization wide.

Thank you.

**Comment by originalfoo** [comment]
No, not yet. Currently I manually just copy the file between existing repos.

**Comment by leonardochaia** [comment]
So, nothing is perfect but this is how I've done it and it works..

For anyone who's wondering..
You need a `.nuspec` file:

```xml
<?xml version="1.0"?>
<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
  <metadata>
    [..]
   <id>YourPackageId</id>
    <developmentDependency>true</developmentDependency>
    <dependencies>
      <dependency id="StyleCop.Analyzers" version="1.1.118" />
      <dependency id="SonarAnalyzer.CSharp" version="8.8.0.18411" />
    </dependencies>
  </metadata>
  <files>
    <file src="../../stylecop.json" target="build/stylecop.json" />
    <file src="../../.editorconfig" target="build/.editorconfig" />

    <file src="YourPackageId.targets" target="build/" />
  </files>
</package>
```

In the .targets file, copy the files next to the .sln file, which I assume is at the root of the project or at least on a higher directory of the .csproj files.

```xml
<Project>
  <!--
  Copies the files to next to the solution file, which should be
  in the root of the project
  -->
  <Target Name="CopyPackage" AfterTargets="PrepareForBuild">
    <Copy
      SourceFiles="$(MSBuildThisFileDirectory)\.editorconfig"
      DestinationFolder="$(SolutionDir)"
    />
    <Copy
      SourceFiles="$(MSBuildThisFileDirectory)\stylecop.json"
      DestinationFolder="$(SolutionDir)"
    />
  </Target>

  <!-- It is needed for stylecop to work -->
  <ItemGroup>
    <AdditionalFiles Include="$(SolutionDir)\stylecop.json" />
  </ItemGroup>
</Project>
```

I have a .csproj for this but you can just use nuget.exe and the nuspec
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.1</TargetFramework>
    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
    <NuspecFile>.nuspec</NuspecFile>
    <NoDefaultExcludes>true</NoDefaultExcludes>
  </PropertyGroup>
</Project>
```

If you're wondering if you can use only the .csproj instead of the nuspec, I tried and run into many issues, if you figure it out do let me know.
